cp22/hpi-fun2/newsort/problem.tex

```diff
@@ -0,0 +1,28 @@
+\input{template.tex}
+
+\begin{document}
+
+\makeheader
+
+Rick and Morty are on a new adventure. This time they landed in a parallel universe and of course, Morty broke Ricks laser gun which is needed ot get back home. To get the money back for the materials for the gun they try to gamble at the stock market. Rick knows that it's a physical law, that the right stocks always go up - and thats where they want to invest. 
+
+You need to figure out if the values of a stock price you receive are increasing, so that Rick and Morty know where to invest. The problem is: In this parallel universe they use the same arabian digits but they don't have the same ordering. In our world we have the ordering $0 < 1 < 2 < 3 < 4 < 5 ...$ but in a parallel universe this ordering could be $6 < 2 < 1 < 4 \dots$ . This ordering does not apply to exponentials, so 11 is still bigger than 1.
+
+Can you tell Rick and Morty if they should invest in a given stock?
+
+\paragraph*{Input}
+
+The first line contains the number of prices $1\leq n<10^9$ of a stock you have followed by the n prices $0<n_i<10^9$. Last you will receive the digit order of the local universe, wich is a permutation of the digits 0-9.
```

Comment:
It is not clear from this sentence that this is actually three different lines. Please rephrase :)

I think you mean `n` should be at most `10^5` here. I also think you should change it to be `\leq 10^5`, since this makes it easier to read. Make sure this is consistent with your test cases.

Use `$n$ prices`. Also, for the range of values here, I think it would be easier to not talk about the price being in some range, but rather that it is a digit string with a maximum length of 9.

It is also unclear from the task whether leading zeros are allowed (or whatever a zero is in the new digit system). I think you even have testcases with this.

Reply:
> Also, for the range of values here, I think it would be easier to not talk about the price being in some range, but rather that it is a digit string with a maximum length of 9.

I'd let it be that way, as this increases the difficulty a bit. This may already push them in the direction to think about a map and so on. But if you insist I can change it :)